### ISSUE TITLE
Raise an error when use categorical_crossentropy for binary classification

### ISSUE THREAD

**Issue by haifeng-jin** [keras-team/keras#17029]
**System information**.

TensorFlow version (you are using): nightly
Are you willing to contribute it (Yes/No) : Yes

**Describe the feature and the current behavior/state**.

When use `categorical_crossentropy` for binary classification,
it does not raise any error, but have all the losses equal to 0 during training.
It may confuse the user that the model has some problem, but actually it just because they accidentally used the wrong loss.
See [this colab](https://colab.research.google.com/drive/12UkksHyPajWrB9d7NiFbPWsN47LXWZ0R?usp=sharing) for more details.

So I propose to raise an error if the shape of `y_true` and `y_pred` are all `(None, 1)`.
Unless this use case makes any sense.

**Will this change the current api? How?**
No.

**Who will benefit from this feature?**
ML Beginners or experienced engineer who accidentally used the wrong loss.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes/no): yes
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing): judge `y_true` and `y_pred` shape and raise error if they are `(None, 1)`.

**Comment by old-school-kid** [comment]
@haifeng-jin 
I am interested in working on this. 
I was thinking maybe we could find the activation of the output layer of the model and if it were `softmax` then `categorical_crossentropy`  loss function is good enough, and `binary_crossentropy` would practically work all the time, although we can do the same for `binary_crossentropy`  and `sigmoid`  activation function.

**Comment by haifeng-jin** [comment]
Based on our discussion in the team, we agreed on raise a warning if the y_true and y_pred are single dimensional vectors.
We prefer no complicated solution to this problem.

**Comment by old-school-kid** [comment]
@haifeng-jin 
So is this issue still open for contribution? Can I work on it?

**Comment by haifeng-jin** [comment]
@old-school-kid Yes. Thank you!
Please create a pull request and mention me.
I will guide you through the rest of the process.
